src/flownet/config_parser/_config_parser.py

```diff
@@ -1757,39 +1754,42 @@ def parse_config(
-            )
-        except KeyError as err:
-            raise ValueError(
-                f"REGION parameter {config.model_parameters.relative_permeability.region_parameter_from_sim_model} "
-                "not found in input simulation model."
-            ) from err
-        _check_if_all_region_priors_defined(
-            config.model_parameters.relative_permeability,
-            unique_satnum_regions,
-            "SATNUM",
-        )
-    else:
-        if config.model_parameters.relative_permeability.regions[0].id is not None:
+    region_parameters: dict = {"equil": "EQLNUM", "relative_permeability": "SATNUM"}
+    unique_regions: dict = {}
+    for reg_param, flow_region_name in region_parameters.items():
+        scheme = getattr(getattr(config.model_parameters, reg_param), "scheme")
+        if scheme not in available_region_schemes:
             raise ValueError(
-                "The region number for the first relative permeability region parameter should not be set, \n"
-                "or set to 'None' when using the 'global' or 'individual' options"
+                f"The {reg_param} scheme "
+                f"'{scheme}' is not valid.\n"
+                f"Valid options are {available_region_schemes}"
             )
-
-    if config.model_parameters.equil.scheme == "regions_from_sim":
-        unique_eqlnum_regions = field_data.get_unique_regions("EQLNUM")
-        _check_if_all_region_priors_defined(
-            config.model_parameters.equil, unique_eqlnum_regions, "EQLNUM"
-        )
-    else:
-        if config.model_parameters.equil.regions[0].id is not None:
-            raise ValueError(
-                "The region number for the first equilibrium region parameter should not be set, or set to 'None'\n"
-                "when using the 'global' or 'individual' options"
+        if scheme == "regions_from_sim":
+            reg_param_sim_model = getattr(
+                getattr(config.model_parameters, reg_param),
+                "region_parameter_from_sim_model",
             )
+            try:
+                unique_regions[reg_param] = field_data.get_unique_regions(
+                    reg_param_sim_model
+                )
+            except KeyError as err:
+                raise ValueError(
+                    f"REGION parameter {reg_param_sim_model} "
+                    "not found in input simulation model."
+                ) from err
+            _check_if_all_region_priors_defined(
+                getattr(config.model_parameters, reg_param),
+                unique_regions[reg_param],
+                flow_region_name,
+            )
+        else:
+            regions = getattr(getattr(config.model_parameters, reg_param), "regions")
+            if regions[0].id is not None:
+                raise ValueError(
+                    f"The region number for the first {reg_param} region parameter should not be set, \n"
```

Comment:
This looks like the exact same error as for the relative permeability.

Potential for re-use of code?
Maybe the user should know for which parameter the error is given?

Reply:
This is inside a for loop where i think i have done what you ask for
```
    region_parameters: dict = {"equil": "EQLNUM", "relative_permeability": "SATNUM"}
    unique_regions: dict = {}
    for reg_param, flow_region_name in region_parameters.items():
```

Reply:
reg_param is "EQLNUM"/"SATNUM" isn't it...? I was thinking to update the warning such that it says:

"For the **relative permeability paramter**, the region number for the first [...]"

We should know that there isn't it? For example the key in the dict should have that info.

Reply:
reg_param is "equil" or "relative_permeability" (the keywords in the config yaml). I could of course make another dictionary that relates "equil"/"relative_permeability" to "equilibrium region"/"relative permeability".

Reply:
No no, that was not the point. "equil" and "relative_permeability" is ✔️ 

But you are not printing that in the warning, isn't it? Or are you?

Reply:
Never mind.